lib/rules/security/compiler-version.js

```diff
@@ -44,6 +44,16 @@ class CompilerVersionChecker extends BaseChecker {
     this.requirement = (config && config.getString(ruleId, DEFAULT_SEMVER)) || DEFAULT_SEMVER
   }
 
+  SourceUnit(node) {
+    const hasPragmaDirectiveDef = node.children.filter(
```

Comment:
`node.children.some` and then `if (!hasPragmaDirective)`.

Reply:
Good catch,done!